firefly/kernel/timer/timer.cpp

```diff
@@ -14,11 +18,23 @@ namespace timer {
 namespace {
 // This will just get increased
 static volatile uint64_t ticks = 0;
+static volatile uint32_t ticks_20ms = 0;
 }  // namespace
 
-void timer_irq() {
-    debugLine << "timer\n"
-              << fmt::endl;
+void timer_irq(Registers* stack) {
+    if (tasks::Scheduler::accessor().getTask() == nullptr) {
```

Comment:
Not a fan of hard coding what to do when a timer irq fires in here...
Ideally each component would register it's own handler, so the scheduler would register it's reschedule function in a timer irq.

We can go over this in the TDD though.